src/Mod/Draft/draftfunctions/move.py

```diff
@@ -215,4 +215,116 @@ def copy_moved_edge(object, edge_index, vector):
         vertex2 = object.Placement.multVec(object.Points[edge_index+1]).add(vector)
     return make_line.make_line(vertex1, vertex2)
 
-## @}
+
+# This part is higly experimental and used for moveSubElements function
+
+def parse_shape(shape, selected_vertexes, vector):
+    """ Parse the given shape and rebuild it according to its
+    original topological structure
+    """
+    import Part
+    if shape.ShapeType == "Compound":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        compound = Part.Compound(new_sub_shapes)
+        return compound
+
+    elif shape.ShapeType == "CompSolid":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        compound = Part.makeCompound(new_sub_shapes)
+
+        return shape
+
+    elif shape.ShapeType == "Solid":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        solid = Part.Solid(new_sub_shapes)
+        return solid
+
+    elif shape.ShapeType == "Shell":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        shell = Part.Shell(new_sub_shapes)
+        return shell
+
+    elif shape.ShapeType == "Face":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        face = Part.Face(new_sub_shapes)
+        return face
+
+    elif shape.ShapeType == "Wire":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        wire = Part.Wire(new_sub_shapes)
+        return wire
+
+    elif shape.ShapeType == "Edge":
+        # to be refined
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        edge = Part.makeLine(new_sub_shapes[0].Point, new_sub_shapes[1].Point)
+        return edge
+
+    elif shape.ShapeType == "Vertex":
+
+        for sv in selected_vertexes:
+            if shape.X == sv.X and shape.Y == sv.Y and shape.Z == sv.Z:
+                return Part.Vertex(App.Vector(shape.X + vector.x, shape.Y + vector.y, shape.Z + vector.z)) 
```

Comment:
Since this is in a loop, here the return is correct. We want to return the `Vertex` as soon as the condition is met. So, we only need two return statements, this one, and the last `return shape`.

Alternatively, we can `break` the loop, and save the `Vertex` in `shape`, so then we only use one `return`.
```py
elif shape.ShapeType == "Vertex":
    for ...:
        if shape.X == sv.X ...:
            shape = Part.Vertex(...)
            break

return shape
```